interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java

```diff
@@ -62,4 +73,36 @@ public void getSeverityTest() {
     public void toStringTest() {
         assertTrue(invalidApiResults.get(0).toString().contains(ERROR_MESSAGE));
     }
-}
\ No newline at end of file
+
+    @Test
+    public void testWithDetails() throws Exception{
+        Position start = new Position(1, 1);
+        Option<String> location = Option.apply(CUSTOM_LOCATION);
+        String message = CUSTOM_ERROR_MESSAGE;
+        ValidationResult validationResult = createAMFValidationResult(message, start, location);
+        ApiValidationResultImpl apiValidationResult = new ApiValidationResultImpl(validationResult);
+
+        String actualMessage = apiValidationResult.getMessage();
+        String expectedMessage = FULL_CUSTOM_ERROR_MESSAGE;
+        assertThat(actualMessage, is(expectedMessage));
+    }
+
+    @Test
+    public void testWithoutDetails() throws Exception{
```

Comment:
add test where location is not empty, and position is 0,0 , for full condition coverage